autoscale/src/main/java/com/sequenceiq/periscope/service/ClusterService.java

```diff
@@ -318,4 +323,11 @@ private void calculateClusterStateMetrics() {
         periscopeMetricService.submit(MetricType.CLUSTER_STATE_SUSPENDED,
                 clusterRepository.countByStateAndAutoscalingEnabledAndPeriscopeNodeId(SUSPENDED, true, periscopeNodeConfig.getId()));
     }
+
+    private void calculateUnprocessedScalingActivityMetrics() {
+        // TODO: Should we fetch a broader collection of scaling events?
+        Collection<Long> activityIds = scalingActivityService.findAllInStatusesThatStartedBefore(ActivityStatus.getStatusesApplicableForCleanup(),
+                UNPROCESSED_SCALING_ACTIVITY_AGE_HOURS);
+        periscopeMetricService.submit(MetricType.STALE_SCALING_ACTIVITY, activityIds.size());
```

Comment:
This metric will be generated only once as part of each deployment/restart. Can you elaborate the reasoning for it?

Reply:
This should be updated periodically as done in `ClusterSerivce`. Refactored this into `ScalingActivityService` instead, following the same pattern